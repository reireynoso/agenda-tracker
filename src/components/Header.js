import React from 'react';
import {useDispatch} from 'react-redux';
import {withRouter} from 'react-router'
import {startLogout} from '../actions/auth';

export const Header = ({history}) => {
    const dispatch = useDispatch();
    return (
        <header id="header">
            <section className="align-center">
                <span id="header__title">Agenda Tracker</span>
                <button id="header__logout" onClick={
                    () => dispatch(startLogout())
                    // () => history.push('/')
                }>Log Out</button>
            </section>
        </header>
    )
}

export default withRouter(Header)