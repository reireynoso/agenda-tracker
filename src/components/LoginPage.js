import React from 'react';
import {useDispatch} from 'react-redux';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import Link from '@material-ui/core/Link';
import Paper from '@material-ui/core/Paper';
import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

import {startLogin} from '../actions/auth'

const Copyright = () => {
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {'Copyright Â© '}
      <Link target="_blank" rel="noopener" color="inherit" href="https://reireynoso.github.io/#/">
        reiReynoso
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}

const useStyles = makeStyles((theme) => ({
  root: {
    height: '100vh',
  },
  image: {
    backgroundImage: 'url(https://source.unsplash.com/random)',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover',
    backgroundPosition: 'center',
  },
  paper: {
    margin: theme.spacing(8, 4),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
    width: "fit-content"
  },
  formContainer: {
      margin: "auto"
  }
}));


export const LoginPage = ({test=() => console.log('testing')}) => {
  const classes = useStyles();

  const dispatch = process.env.NODE_ENV === "test" ? test : useDispatch();

  return (
    <Grid container component="main" className={classes.root}>
      <CssBaseline />
      <Grid item xs={false} sm={4} md={7} className={classes.image} />
      <Grid className={classes.formContainer} item xs={12} sm={8} md={5} component={Paper} elevation={6} square>
        <div className={classes.paper}>
            <Avatar className={classes.avatar}>
                <LockOutlinedIcon />
            </Avatar>
            <Typography component="h1" variant="h5">
                Agenda Tracker
            </Typography>
            <Button
                type="submit"
                fullWidth
                variant="contained"
                color="primary"
                className={classes.submit}
                onClick={() => dispatch(startLogin())}
                // onClick={() => history.push('/dashboard')}
            >
                Login w/ Google
            </Button>
            <Box mt={5}>
                <Copyright />
            </Box>
        </div>
      </Grid>
    </Grid>
  );
}

export default LoginPage