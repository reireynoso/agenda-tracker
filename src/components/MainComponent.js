import React from 'react';
import {useSelector, useDispatch} from 'react-redux'
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import {DragDropContext, Droppable, Draggable} from 'react-beautiful-dnd'

import {changePositioning} from '../actions/todosAction'

export default () => {
    const tasks = useSelector(state => state.tasks)
    // console.log(tasks)
    const columns = useSelector(state => state.columns) 

    const dispatch = useDispatch()
    const useStyles = makeStyles(() => ({
        root: {
          flexGrow: 1
        },
        paper: {
          padding: "0.6rem",
          border: "1px solid white",
          marginBottom: "5px",
          background: "#1c88bf",
          cursor: "pointer"
        },
        column: {
            minHeight: "260px",
            padding: "0.5rem"
        }
      }));
      
    const classes = useStyles();

    const onDragEnd = result => {
      // console.log('drag end')
      const {destination, source} = result;
      //if there's no destination do nothing
      if(!destination){
        return;
      }
      //if it was dropped in the same exact location, no need to do anything
      if(
        destination.droppableId === source.droppableId && destination.index === source.index
      ){
        return;
      }    
      dispatch(changePositioning(result))
    }
    
    return(
        <div className="align-center">
            <main id="main">
                <h1 id="main__title">Agendas</h1>
                  <DragDropContext
                    onDragEnd={onDragEnd}
                  >
                    <Grid justify="center" container className={classes.root} spacing={2}>  
                          {Object.keys(columns).map(key => (
                          <Grid key={key} item xs={8} sm={6} md={4}>
                            <Droppable key={key} droppableId={key}>
                              {
                                provided => (    
                                    <Paper innerRef={provided.innerRef} {...provided.droppableProps} className={classes.column}>
                                      <h2 className="category__title">{columns[key].title}</h2>
                                      {
                                          columns[key].taskIds.map((taskid,index) => {
                                              return <Draggable
                                                key={taskid}
                                                draggableId={taskid}
                                                index={index}
                                              >
                                                {
                                                  (provided) => (
                                                    <Paper {...provided.draggableProps} {...provided.dragHandleProps} innerRef={provided.innerRef} key={taskid} className={classes.paper}>{tasks[taskid].content}</Paper>
                                                  )
                                                }
                                                
                                              </Draggable> 
                                          })
                                      }
                                      {
                                        provided.placeholder
                                      }
                                    </Paper>
                                )
                              }
                              </Droppable>
                          </Grid>
                          ))}       
                    </Grid>
                  </DragDropContext>
            </main>
        </div>
    )
}