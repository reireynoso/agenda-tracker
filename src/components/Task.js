import React from 'react';
import {useSelector, useDispatch} from 'react-redux'
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import {Draggable} from 'react-beautiful-dnd'
import {removeTask} from '../actions/todosAction'
// import ContentEditable from 'react-contenteditable'

export default ({taskid,index,col}) => {
    const dispatch = useDispatch()
    const tasks = useSelector(state => state.tasks);

    const useStyles = makeStyles(() => ({
        paper: {
            padding: "0.6rem",
            border: "1px solid white",
            marginBottom: "5px",
            background: "#1c88bf",
            cursor: "pointer"
        }
    }));

    const classes = useStyles();

    const handleRemoveTask = (e) => {
        e.preventDefault();
        // console.log(col, taskid)
        dispatch(removeTask({
            col,
            taskid
        }))
    }
    return (
        <Draggable
        key={taskid}
        draggableId={taskid}
        index={index}
        >
        {
            (provided) => (
            <Paper 
            {...provided.draggableProps} 
            {...provided.dragHandleProps} 
            innerRef={provided.innerRef} 
            key={taskid} 
            className={classes.paper}
            >   
                {
                    // <div className="task__content">
                    //     <ContentEditable
                    //         className = "task__content-edit"
                    //         html={
                    //             `<span>${tasks[taskid].content}</span>`
                    //         }
                    //     />
                    // </div>
                }
                    <span className="task__content">{tasks[taskid].content}</span>
                {
                    col === "column-3" && <button onClick={handleRemoveTask} className="task__delete-button">X</button>
                }
            </Paper>
            )
        }
        </Draggable> 
    )
}