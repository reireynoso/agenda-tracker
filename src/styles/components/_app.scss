#app {
    height: 100vh;
    // width: 100vw;
    // margin: 0 7rem;
    // border-left: 1px solid $off-white;
    // border-right: 1px solid $off-white;

    display: flex;
    flex-direction: column;
}