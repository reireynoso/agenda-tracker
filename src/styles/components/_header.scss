#header {
    height: 5rem;
    background: $off-white;
    color: $dark-grey;
    display: flex;

    .align-center {
        display: flex;
        align-items: center;
        justify-content: space-between;
    }

}

#header__title {
    font-size: $font-size-small;
}

#header__logout {
    font-size: $font-size-xsmall;
    border: none;
    background: none;
    padding: 5px;
    outline: none;
}
