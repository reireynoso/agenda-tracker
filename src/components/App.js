import React from 'react';
import Header from './Header';
import MainComponent from './MainComponent';

export default () => {
    return (
        <main id="app">
            <Header/>
            <MainComponent/>
        </main>
    )
}