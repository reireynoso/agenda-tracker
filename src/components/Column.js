import React, {useState} from 'react';
import {useSelector, useDispatch} from 'react-redux'
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Input from '@material-ui/core/Input'
import Grid from '@material-ui/core/Grid';
import {Droppable} from 'react-beautiful-dnd'
import {v4 as uuidv4} from 'uuid'

import Task from '../components/Task'
import {addNewTask} from '../actions/todosAction'
// import {startAddNewTask} from '../actions/todosAction'

export default ({col}) => {
    const dispatch = useDispatch();
    const columns = useSelector(state => state.todosReducer.columns);
    const [task, setTask] = useState("")
    const useStyles = makeStyles(() => ({
        column: {
            minHeight: "260px",
            padding: "0.5rem"
        },
        input: {
            width: "100%",
            '&::after': {
                borderBottom: "2px solid #1c88bf"
            }
        }
    }));

    const classes = useStyles();

    const handleAddNewTask = (e) => {
        if(e.key === "Enter"){
            if(task){
                const newTask = {
                    id: uuidv4(),
                    content: task
                }
                // console.log(`create`, task)
                dispatch(addNewTask(newTask))
                // dispatch(startAddNewTask(newTask, initial))
                setTask("")
            }
        }
    }

    return (
        <Grid  item xs={8} sm={6} md={4}>
        <h2 className="category__title">{columns[col].title}</h2>
        <Droppable droppableId={col}>
            {
            provided => (    
                <Paper 
                innerRef={provided.innerRef} 
                {...provided.droppableProps} 
                className={classes.column}>        
                    {
                        columns[col].taskIds && columns[col].taskIds.map((taskid,index) => {
                            return <Task col={col} key={taskid} taskid={taskid} index={index}/>
                        })
                    }
                    {
                    provided.placeholder
                    }
                    {
                    col === "column-1" ? <Input 
                        value={task}
                        className={classes.input}
                        placeholder="New task. Press Enter"
                        onChange = {(e) => setTask(e.target.value)}
                        onKeyPress={handleAddNewTask}
                        /> : null
                    }
                </Paper>
            )
            }
            </Droppable>
        </Grid>
    )
}