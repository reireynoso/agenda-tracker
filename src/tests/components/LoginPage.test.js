import React from 'react';
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'
import {LoginPage} from '../../components/LoginPage'
import configureStore from 'redux-mock-store'
import {Provider} from 'react-redux'

test("Render Login page", () => {
    const mockStore = configureStore();
    let store = mockStore({})
    const component = <Provider store={store}>
        <LoginPage/>
    </Provider>
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot()
})