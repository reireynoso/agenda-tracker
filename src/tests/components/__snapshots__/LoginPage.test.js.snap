// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login page 1`] = `
<main
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container"
>
  <div
    className="MuiGrid-root makeStyles-image-2 MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-7"
  />
  <div
    className="MuiPaper-root MuiGrid-root makeStyles-formContainer-7 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 MuiPaper-elevation6"
  >
    <div
      className="makeStyles-paper-3"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
      <h1
        className="MuiTypography-root MuiTypography-h5"
      >
        Agenda Tracker
      </h1>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-6 MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Login w/ Google
        </span>
      </button>
      <div
        className="MuiBox-root MuiBox-root-8"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
        >
          Copyright Â© 
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
            href="https://reireynoso.github.io/#/"
            onBlur={[Function]}
            onFocus={[Function]}
            rel="noopener"
            target="_blank"
          >
            reiReynoso
          </a>
           
          2020
          .
        </p>
      </div>
    </div>
  </div>
</main>
`;
